docs/src/model_setup/output_writers.md

```diff
@@ -158,8 +158,46 @@ NetCDFOutputWriter scheduled on IterationInterval(1):
 └── file size: 17.8 KiB
 ```
 
-`NetCDFOutputWriter` also accepts an arbitrary `grid` for input that can be used to interpolate or subset the output fields. The output `grid` must be in the same architecture as the original grid.
+`NetCDFOutputWriter` can also be configured for `outputs` that are interpolated or regridded to a different grid than `model.grid`. To use this functionality, the `output_grid` must be passed explicitly when constructing `NetCDFOutputWriter` along with the regridded / interpolated `outputs`.
 
+```jldoctest
+using Oceananigans
+
+grid = RectilinearGrid(size = (1, 1, 8), extent = (1,1,1));
+model = NonhydrostaticModel(; grid, closure = ScalarDiffusivity(ν=1e-2))
+
+set!(model, u=(x, y, z,) -> z)
+
+simulation = Simulation(model,
+                        Δt=0.5*maximum(zspacings(grid, Center())) / maximum(abs, model.velocities.u),
+                        stop_time=20)
+
+simulation.output_writers[:fullfields] = NetCDFOutputWriter(model, (; model.velocities.u),
+                                                            filename = "fullfields.nc",
+                                                            schedule = TimeInterval(5),
+                                                            overwrite_existing = true,)
+
+coarse_grid = RectilinearGrid(size = (grid.Nx, grid.Ny, grid.Nz÷2), extent = (grid.Lx, grid.Ly, grid.Lz))
+coarse_u = Field{Face, Center, Center}(coarse_grid)
+
+using Oceananigans.Fields: interpolate!
+update_coarse_u(simulation) = interpolate!(coarse_u, simulation.model.velocities.u)
+simulation.callbacks[:update_interp] = Callback(update_coarse_u)
+
+simulation.output_writers[:coarsefields] = NetCDFOutputWriter(model, (; coarse_u,), coarse_grid;
+                                                              filename="coarsefields.nc",
+                                                              schedule=TimeInterval(5),
+                                                              overwrite_existing=true,)
```

Comment:
Nice work! Here's a few suggestions:

* don't include outputting the original fields. We have other docstrings for that.
* Make the example simpler. Don't set initial conditions, don't compute `dt`, don't put in a random stop time like "20". `stop_iteration=1` is fine.
* I recomend an output named tuple like `(; u = coarse_u)`. In other words, the user wil lbe able to tell from the filename and grid that the output is coarse. We don't need to additionally name the output "`coarse_u`".
* Make sure to use correct style with spaces after `=` etc

It will make the script shorter to write

```julia
interpolate_u(model) = interpolate!(coarse_u, model.velocities.u)
outputs = (; u = interpolate_u)
```

in other words, make the output a function itself. Then you don't have to use a callback to compute the output.

Reply:
Thanks! about the spaces. I usually do that, but got confused because there are no spaces after `=` for the other examples in the docs. Should I take the time and correct the other examples?

Reply:
@glwagner 
If I try what you suggested and use the function directly in the `outputs`, it returns
```julia
Custom output u needs dimensions!
```

Reply:
I'll fix that.

Reply:
Okay put in a feature that helps with user functions that return fields (the same feature already existed in JLD2OutputWriter after a fact, and I didn't realize it was missing for netcdf)

Reply:
see if you get the same error now

Reply:
I still got the same error. Is that anything I could share to help u debug it?

Reply:
Yes, a MWE would help.

Reply:
It is exactly the same code from the example.
Let me know if I am doing something wrong. What I am doing is, in a test directory, I am running

```bash
rm -f *.toml && touch Project.toml && julia --project -e 'using Pkg; Pkg.add(url = "https://github.com/CliMA/Oceananigans.jl/#tc/grid-to-netcdfwriter")'
```
(I also tried to run with `Pkg.add(path = "../Oceananigans.jl")` , which is the directory I have the code at the current branch)

Then 

```bash
julia --project code.jl
```

`code.jl` contains the following code

```julia
using Oceananigans

grid = RectilinearGrid(size = (1, 1, 8), extent = (1, 1, 1));
model = NonhydrostaticModel(; grid, closure = ScalarDiffusivity(ν = 1e-2))

set!(model, u = (x, y, z,) -> z)

simulation = Simulation(model, Δt = 1, stop_iteration = 1)

coarse_grid = RectilinearGrid(size = (grid.Nx, grid.Ny, grid.Nz÷2), extent = (grid.Lx, grid.Ly, grid.Lz))
coarse_u = Field{Face, Center, Center}(coarse_grid)

using Oceananigans.Fields: interpolate!

# # this works
# update_coarse_u(simulation) = interpolate!(coarse_u, simulation.model.velocities.u)
# simulation.callbacks[:update_interp] = Callback(update_coarse_u)

# outputs = (; u = coarse_u,)

# this doesn't work
interpolate_u(model) = interpolate!(coarse_u, model.velocities.u)
outputs = (; u = interpolate_u)

simulation.output_writers[:coarse_u] = NetCDFOutputWriter(model, outputs, coarse_grid;
                                                              filename = "coarse_u.nc",
                                                              schedule = TimeInterval(5),
                                                              overwrite_existing = true,)
```

which returns 

```julia
Custom output u needs dimensions!

Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] define_output_variable!(dataset::NCDatasets.NCDataset{Nothing, Missing}, output::Nothing, name::String, array_type::Type, deflatelevel::Int64, output_attributes::Dict{String, String}, dimensions::Dict{Any, Any})
   @ Oceananigans.OutputWriters ~/.julia/packages/Oceananigans/r2jWy/src/OutputWriters/netcdf_output_writer.jl:460
 [3] initialize_nc_file!(filepath::String, outputs::Dict{String, typeof(interpolate_u)}, .... .... 
```

Reply:
Ok fixed the issue (it was with `interpolate!`. Also I improved the readability of output writer code. I don't think custom output is really properly supported right now. There's some kind of assumption about the array type for custom output taking place which would prevent people from doing non-array output.

Here's a more minimal example that you can use in the docstring:

```julia
using Oceananigans
using Oceananigans.Fields: interpolate!

grid = RectilinearGrid(size=(1, 1, 8), extent=(1, 1, 1));
model = NonhydrostaticModel(; grid)
simulation = Simulation(model, Δt=1, stop_iteration=1)

coarse_grid = RectilinearGrid(size=(grid.Nx, grid.Ny, grid.Nz÷2), extent=(grid.Lx, grid.Ly, grid.Lz))
coarse_u = Field{Face, Center, Center}(coarse_grid)

interpolate_u(model) = interpolate!(coarse_u, model.velocities.u)
outputs = (; u = interpolate_u)

rm("coarse_u.nc", force=true)

simulation.output_writers[:coarse_u] = NetCDFOutputWriter(model, outputs, coarse_grid;
                                                          filename = "coarse_u.nc",
                                                          schedule = IterationInterval(1))
```

I got rid of everything that wasn't strictly necessary for the example. The problem with including "random" extra stuff like a closure to the model is that it can be confusing for a new user, whether or not those things are crucial to the example or not.